Add raw token info in API - Getting started and remove v4 string from API requests (#2804)
wazuh/wazuh-documentation: source/user-manual/api/getting-started.rst

Wazuh API endpoints require authentication in order to be used. Therefore, all calls must include a JSON Web Token. JWT is an open standard (RFC 7519) that defines a compact and self-contained way for securely transmitting information between parties as a JSON object. Follow the steps below to log in and obtain a token in order to run any endpoint:

> 1. Use the cURL command to log in, the API will provide a JWT token upon success. Replace <user> and <password> with yours. By default, the user is ``wazuh`` and the password is ``wazuh``. If ``SSL`` (https) is enabled in the API and it is using the default **self-signed certificates**, it will be necessary to add the parameter ``-k``. Use the ``raw`` option to get the token in a plain text format, this way, we will be able to export the token to an environment variable and make API requests in an easier way. Using ``raw=true`` is highly recommended as tokens could be long and difficult to use.

Reviewer: ```suggestion
1. Use the cURL command to log in, the API will provide a JWT token upon success. Replace <user> and <password> with yours. By default, the user is ``wazuh`` and the password is ``wazuh``. If ``SSL`` (https) is enabled in the API and it is using the default **self-signed certificates**, it will be necessary to add the parameter ``-k``. Use the ``raw`` option to get the token in a plain text format. Querying the login endpoint with ``raw=true`` is highly recommended when using cURL commands as tokens could be really long and difficult to handle otherwise. Exporting the token to an environment variable will ease the use of API requests after login.
```

    .. code-block:: console

>         # curl -u <user>:<password> -k -X GET "https://localhost:55000/security/user/authenticate?raw=true"

Reviewer: Delete

    You will get a result similar to the following.

    .. code-block:: none
        :class: output

        <YOUR_JWT_TOKEN>

> 

Reviewer: Delete

>     Export the token to an environment variable so API requests will be easier to make:

Reviewer: ```suggestion
    Export the token to an environment variable to use it in authorization header of future API requests:
```

    .. code-block:: console

        # TOKEN=$(curl -u <user>:<password> -k -X GET "https://localhost:55000/security/user/authenticate?raw=true")

>     If you set ``raw=false`` or you leave it by default, you will get a result similar to the following. Copy the token that you will find in ``<YOUR_JWT_TOKEN>`` without the quotes.

Reviewer: ```suggestion
    By default (``raw=false``), the token is obtained in an ``application/json`` format. If using this option, copy the token that you will find in ``<YOUR_JWT_TOKEN>`` without the quotes.
```

            "timestamp": "2020-05-25T07:05:00+0000"
        }

> Once we are logged in we can run any endpoint following the structure below. Please, do not forget to replace <endpoint> with your own value. In case you are not using the environment variable, replace $TOKEN with your jwt token.

Reviewer: ```suggestion
Once logged in, it is possible to run any endpoint following the structure below. Please, do not forget to replace <endpoint> with your own value. In case you are not using the environment variable, replace $TOKEN with your jwt token.
```

.. code-block:: console

    # curl -k -X <METHOD> "https://localhost:55000/<ENDPOINT>" -H  "Authorization: Bearer $TOKEN"

Logging into the API via scripts
--------------------------------

> With these scripts, it will be easier to understand the two ways of logging into the API: using `raw=true` or `raw=false`.

Reviewer: ```suggestion
The following scripts provide API login examples using default (`false`) or plain text (`true`) `raw` parameter. They intend to bring the user closer to real use cases with Wazuh API.
```

> 1. Logging in with Python and non-raw token:

Reviewer: ```suggestion
1. Logging in with Python:
```

.. code-block:: python

    #!/usr/bin/env python3

    import json
    import requests
    import urllib3
    from base64 import b64encode

    # Disable insecure https warnings (for self-signed SSL certificates)
    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

    # Configuration
    protocol = 'https'
    host = 'localhost'
    port = 55000
    user = 'wazuh'
    password = 'wazuh'

>     login_endpoint = 'security/user/authenticate?raw=false'

Reviewer: ```suggestion
    login_endpoint = 'security/user/authenticate'
```

    login_url = f"{protocol}://{host}:{port}/{login_endpoint}"
    basic_auth = f"{user}:{password}".encode()
    login_headers = {'Content-Type': 'application/json',
                     'Authorization': f'Basic {b64encode(basic_auth).decode()}'}

>     print("\n- Getting token ...\n")

Reviewer: ```suggestion
    print("\nLogin request ...\n")
```

    response = requests.get(login_url, headers=login_headers, verify=False)
    token = json.loads(response.content.decode())['token']

> 

Reviewer: Print token response as answer to login stage

>     # new authorization with the token we got

Reviewer: ```suggestion
    # New authorization header with the JWT token we got
```

    requests_headers = {'Content-Type': 'application/json',
                        'Authorization': f'Bearer {token}'}

    print("\n- API calls with TOKEN environment variable ...\n")

>     print("Getting default information:\n")

Reviewer: ```suggestion
    print("Getting API information:\n")
```

    response = requests.get(f"{protocol}://{host}:{port}/?pretty=true", headers=requests_headers, verify=False)
    print(response.text)

>     print("\nGetting /agents/summary/os:\n")

Reviewer: ```suggestion
    print("\nGetting agents status summary:\n")
```

    response = requests.get(f"{protocol}://{host}:{port}/agents/summary/status?pretty=true", headers=requests_headers, verify=False)
    print(response.text)

    print("\n\nEnd of the script.\n")

> With this script, you will get a result similar to the following:

Reviewer: ```suggestion
Running the script provides a result similar to the following:
```

.. code-block:: console

>     # root@wazuh-master:/# python3 login_script.py

Reviewer: Update results with the changes to the script

    - Getting token ...

    - API calls with TOKEN environment variable ...

    Getting default information:

    {
       "title": "Wazuh API REST",
       "api_version": "4.0.0",
       "revision": 4000,
       "license_name": "GPL 2.0",
       "license_url": "https://github.com/wazuh/wazuh/blob/master/LICENSE",
       "hostname": "wazuh-master",
       "timestamp": "2020-08-18T08:35:45+0000"
    }

    Getting /agents/summary/os:

    {
       "data": {
          "active": 9,
          "disconnected": 2,
          "never_connected": 2,
          "pending": 0,
          "total": 13
       }
    }

    End of the script.

> 2. Logging in with a bash script and raw token:

Reviewer: Update stages accordingly to python script changes

.. code-block:: bash

    #!/bin/bash

    echo -e "\n- Getting token...\n"

    TOKEN=$(curl -u wazuh:wazuh -k -X GET "https://localhost:55000/security/user/authenticate?raw=true")

    echo -e "\n- API calls with TOKEN environment variable ...\n"

    echo -e "Getting default information:\n"

    curl -k -X GET "https://localhost:55000/?pretty=true" -H  "Authorization: Bearer $TOKEN"

    echo -e "\n\nGetting /agents/summary/os:\n"

    curl -k -X GET "https://localhost:55000/agents/summary/status?pretty=true" -H  "Authorization: Bearer $TOKEN"

    echo -e "\n\nEnd of the script.\n"

> With this script, you will get a result similar to the following:

Reviewer: ```suggestion
Running the script provides a result similar to the following:
```
